internal/k8s/sync.go

```diff
@@ -230,8 +232,7 @@ func (client *Client) GetUser(email string) (UserResource, error) {
 
 func (client *Client) AssertPassword(username string, password string) bool {
 	logger := misc.GetLogger()
-	username = strings.Replace(username, "@", "__at__", 1)
-	userCR, _ := getCRForGroupKindWithLabel("iam-backbone.org", "backboneuser", "email", username, client.k8sDynClient)
+	userCR, _ := getCRForGroupKindWithUsername("iam-backbone.org", "backboneuser", username, client.k8sDynClient)
```

Comment:
It's probably better to decouple the k8s abstraction from the business logic here (and everywhere else, but that's for another time), in other words, delete this getCRForGroupKindWithLabel() method and instead in the  AssertPassword(), create a private method that filters CRs by username.
Call getCRsForGroupKind("iam-backbone.org", "backboneuser"...) , apply the filter, convert and return the User model.

Reply:
I'm not sure that I fully understand.
Do you want to remove the cr fetch logic from the AssertPassword completely? and pass the CR from outside to it as a parameter?

and when u write create private method as in on the Client struct or just as it is now (getCRForGroupKindWithUsername).

I can change the logic inside of it to call getCRsForGroupKind and filter based on username

Reply:
Added the method and did a small refactor in the labels method for checking the length of crs list